pkg/tanka/workflow.go

```diff
@@ -90,6 +90,54 @@ func Diff(baseDir string, mods ...Modifier) (*string, error) {
 	return kube.Diff(l.Resources, opts.diff)
 }
 
+// Delete parses the environment at the given directory (a `baseDir`) and deletes
+// the generated objects from the Kubernetes cluster defined in the environment's
+// `spec.json`.
+func Delete(baseDir string, mods ...Modifier) error {
+	opts := parseModifiers(mods)
+
+	l, err := load(baseDir, opts)
+	if err != nil {
+		return err
+	}
+	kube, err := l.connect()
+	if err != nil {
+		return err
+	}
+	defer kube.Close()
+
+	// The list of objects is sorted so that it applies correctly; to delete it, we need to reverse the order.
+	reversed := manifest.List{}
+	for i := len(l.Resources) - 1; i >= 0; i-- {
+		reversed = append(reversed, l.Resources[i])
+	}
+
+	// show diff
+	diff, err := kubernetes.StaticDiffer(false)(reversed)
+	switch {
+	case err != nil:
+		// This is not fatal, the diff is not strictly required
+		fmt.Println("Error diffing:", err)
+	case diff == nil:
+		tmp := "Warning: There are no differences. Your apply may not do anything at all."
+		diff = &tmp
+	}
+
+	// in case of non-fatal error diff may be nil
+	if diff != nil {
+		b := term.Colordiff(*diff)
+		fmt.Print(b.String())
+	}
+
+	// prompt for confirmation
+	if opts.delete.AutoApprove {
```

Comment:
You pass `withApplyAutoApprove`, but refer to deleteOpts (which you never seem to modify) here.

As deleteOpts is an alias type of applyopts, I guess you can just remove it here and cast applyOpts

Reply:
Fixed.